dataprofiler/profilers/utils.py

```diff
@@ -245,3 +245,33 @@ def overlap(x1, x2, y1, y2):
             (y1 <= x2 <= y2) or
             (x1 <= y1 <= x2) or
             (x1 <= y2 <= x2))
+
+def add_nested_dictionaries(first_dict, second_dict):
+    """
+    Merges two dictionaries together and adds values together
+
+    :param first_dict: dictionary to be merged
+    :type first_dict: dict
+    :param second_dict: dictionary to be merged
+    :type second_dict: dict
+    :return: merged dictionary
+    """
+    merged_dict = {}
+    if isinstance(first_dict, collections.defaultdict):
+        merged_dict = collections.defaultdict(first_dict.default_factory)
+
+    for item in first_dict:
+        if item in second_dict:
+            if isinstance(first_dict[item], (dict, collections.defaultdict)):
```

Comment:
can this just be `isinstance(first_dict[item], dict)` ? cause isn't: `isinstance(collections.defaultdict(int), dict) == True`?